### ISSUE TITLE
'HDF Error' when subsetting file

### ISSUE THREAD

**Issue by dopplershift** [Unidata/thredds#330]
The attached file yields:

```
NetCDF: HDF error
```

It looks like #301 on the surface, but the fix for that does not fix this problem. 

This query string allowed me to reproduce locally (my first attempt):

```
COMPRESS_LEV2_20140201000000-GLOBCURRENT-L4-CURekm_15m-ERAWS_EEM-v02.0-fv01.0.nc?var=eastward_ekman_current_velocity&north=79.8750&west=-140&east=179.8750&south=-79.8750&horizStride=1&time_start=2014-02-01T00%3A00%3A00Z&time_end=2014-02-01T00%3A00%3A00Z&timeStride=1&accept=netcdf4
```

**Comment by dopplershift** [comment]
[COMPRESS_LEV2_20140201000000-GLOBCURRENT-L4-CURekm_15m-ERAWS_EEM-v02.0-fv01.0.nc](https://files.zenhub.io/566b5c0e9b2939d368c188f6)
Uploaded using [**ZenHub.io**](http://www.zenhub.io)

**Comment by JohnLCaron** [comment]
It reads ok in toolsUI.

seems to write ok in Writer tab, though i didnt subset. (5.0).

 I assume you get the error from ncss when writing to netcdf4?

**Comment by dopplershift** [comment]
Even have the problem on 5.0 asking for netcdf4, I get:
```IOException sending File NetCDF: HDF error```

Here's the query:
```
COMPRESS_LEV2_20140201000000-GLOBCURRENT-L4-CURekm_15m-ERAWS_EEM-v02.0-fv01.0.nc?var=eastward_ekman_current_velocity&north=79.8750&west=-140&east=170&south=-79.8750&horizStride=1&time_start=2014-02-01T00%3A00%3A00Z&time_end=2014-02-01T00%3A00%3A00Z&timeStride=1&accept=netcdf4
```

**Comment by JohnLCaron** [comment]
i have a fix for this, do you want it on 4.6? obviously needs to be run
through unit tests, so depends if you have already started the release

**Comment by cwardgar** [comment]
Nope, haven't finalized 4.6.4 yet (barely). So yeah, go ahead and submit a PR. You need a full Jenkins run for this, not just Travis?

**Comment by JohnLCaron** [comment]
on second look, its not as easy to fix on 4.6.
so im leaving it off for now.
will fix on 5.0, retrofit later if called for.

**Comment by JohnLCaron** [comment]
Ryan, can you add that file to cdmUnitTest/formats/netcdf4 ? thanks

**Comment by dopplershift** [comment]
File is added.

**Comment by lesserwhirls** [comment]
@JohnLCaron - would it be a pretty straightforward fix in 4.6 following the changes you made in 5?

**Comment by lesserwhirls** [comment]
Just need a sanity check to see if this is still fixed on 5.0

**Comment by cwardgar** [comment]
I've confirmed that this is still fixed on 5.0.
